### ISSUE TITLE
Add workspaces and other parameters to the consumer decorator

### ISSUE THREAD

**Issue by msouff** [tethysplatform/tethys#856]
Right now the consumer only accepts name, url, and regex. It would be nice to have it accept some of the other parameters that the controller decorator accepts.

The problem is that consumers are classes and the way the controller decorator works, it is just providing a wrapper to our other already exisiting decorators such as the @app_workspace one. In the case of @app_workspace, it does not work with consumers because 1) they are classes not functions, and 2) consumers do not have a request, which is needed to determine the active app and therefore its workspace.

We have a few alternatives to explore here when/if we decide to look at this.

**Comment by sdc50** [comment]
I considered doing this when I was developing the `consumer` decorator. I think what held me up is that it would require that every method in the class accept the workspace arguments. I thought it would be better to just use the workspace decorators directly on the methods that you want to have those arguments.

**Comment by swainn** [comment]
I made the same suggestion @sdc50 , but I think the decorators require the method/function to have a Request object passed as an argument. We have the `@with_workspaces` decorator that works with the Bokeh handlers, but nothing that works for  basic Consumers.

**Comment by sdc50** [comment]
That's right. I was confusing the `consumer` decorator with decorating class-based controllers. I'm planning to work with consumers to fix the bokeh-django integration, so maybe I'll get some ideas while delving into that.
